_posts/2021-01-11-howto-nuxeo-package.md

```diff
@@ -0,0 +1,245 @@
+   create nuxeo-acme-package/src/main/resources/install.xml
+   create nuxeo-acme-package/src/main/resources/package.xml
+   create Your project is ready!
+     info You can start editing code or you can continue with calling another generator (nuxeo bootstrap [<generator>..])
+$
+
+```
+
+With those commands, we just created an empty Nuxeo package that installs its own `marketplace-acme` template. We can now configure some specific application configuration by
+editing the file in `nuxeo-acme-package/src/main/resources/install/templates/marketplace-acme/nuxeo.defaults`. For instance we can configure the prefix of the emails sent
+by the platform by adding:
+
+```
+nuxeo.notification.eMailSubjectPrefix="[ACME] "
+```
+
+The package can simply be built by running:
+```shell
+$ mvn clean package
+```
+
+
+Running The Package In A Docker Container
+=========================================
+
+In order to test our package, we will run it inside a Docker container. For this to be reproducible, we will create a dedicated image using the following `Dockerfile` at the root of our project:
+
+```dockerfile
+FROM docker.packages.nuxeo.com/nuxeo/nuxeo:11.4
+
+ARG NUXEO_CLID
+
+COPY nuxeo-acme-package/target/nuxeo-acme-package-1.0-SNAPSHOT.zip /package.zip
+RUN /install-packages.sh --clid ${NUXEO_CLID} /package.zip nuxeo-web-ui
+
+```
+
+To build and run it:
+
+```shell
+$ docker run --rm -i docker.packages.nuxeo.com/nuxeo/nuxeo:11.4 nuxeoctl register && cat /var/lib/nuxeo/instance.clid | sed ':a;N;$!ba;s/\n/--/g' | sed s/--$//
+XXXXX
+$ export NUXEO_CLID=XXXXX
+$ docker build -t nuxeo-acme --build-arg NUXEO_CLID .
+$ docker run -d --name nuxeo nuxeo-acme
+425d138bc6d9b49dee92791296a3a60ec92718dd6b066c9f2821b03e24a6d093
+$ docker exec nuxeo cat /opt/nuxeo/server/nxserver/config/notification-config.xml
+<?xml version="1.0"?>
+<component name="org.nuxeo.ecm.platform.ear.config.notification">
+
+  <extension
+    target="org.nuxeo.ecm.platform.ec.notification.service.NotificationService"
+    point="generalSettings">
+    <settings>
+      <serverPrefix>http://localhost:8080/nuxeo/</serverPrefix>
+      <eMailSubjectPrefix>"[ACME] " </eMailSubjectPrefix>
+      <mailSessionJndiName>java:comp/env/Mail</mailSessionJndiName>
+    </settings>
+  </extension>
+
+</component>
+$ docker rm -f nuxeo
+```
+
+We can see that our configuration has been taken into account.
+
+
+Specifying Another Email Prefix For The DEV Environment
+=======================================================
+
+[ACME] is the default prefix. But for a development environment, it may be a problem that the prefix is the same than in production: it could lead to misinterpreation on some test messages. To change that configuration in dev, we could ask the devops team to manage that configuration at the infrastructure level and override it in the global `nuxeo.conf`. The other and better way of doing it is to add a `nuxeo.dev` file next to the `nuxeo.defaults` one and fill it with:
```

Comment:
```suggestion
[ACME] is the default prefix. But for a development environment, it may be a problem that the prefix is the same than in production: it could lead to misinterpretation on some test messages. To change that configuration in dev, we could ask the devops team to manage that configuration at the infrastructure level and override it in the global `nuxeo.conf`. The other, and better, way of doing it is to add a `nuxeo.dev` file next to the `nuxeo.defaults` one and fill it with:
```